public/partials/demand_new.html

```diff
@@ -20,13 +21,19 @@
                 No matches found for "{{searchText}}".
             </md-not-found>
         </md-autocomplete>
+
+        <md-input-container ng-if="edit_demand">
```

Comment:
again... the use of `ng-if`.  Try and place both elements in a DIV and place the `ng-if` attribute on the div.